### PR TITLE
chore(deps): mass update

### PR DESCRIPTION
**Pull request by miketheman**
We're stuck waiting on a dependabot-core release that includes a fix to
the version of pip-tools that is used, so perform a manual upgrade.

Picks up botocore/boto3 deps that have been excluded from dependabot, so
that's nice.

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

---

* Closes #12898
* Closes #12899
* Closes #12904
* Closes #12907
* Closes #12908
* Closes #12909
* Closes #12928
* Closes #12937
* Closes #12941
* Closes #12942
* Closes #12950
* Closes #12954
* Closes #12956
* Closes #12958
* Closes #12967
* Closes #12974
* Closes #12976
* Closes #12977
* Closes #12988
* Closes #12989
* Closes #12990
* Closes #12991
* Closes #12992
* Closes #12998
* Closes #12999
* Closes #13007
* Closes #13015
* Closes #13016
* Closes #13017
* Closes #13023
* Closes #13024
* Closes #13025
* Closes #13027
* Closes #13028
* Closes #13029
* Closes #13030
* Closes #13032
* Closes #13034
* Closes #13035
* Closes #13037
* Closes #13038
* Closes #13039
* Closes #13040
* Closes #13043
* Closes #13044
* Closes #13045
* Closes #13046
* Closes #13050
* Closes #13051
* Closes #13063
* Closes #13064
* Closes #13065
* Closes #13066
* Closes #13067
* Closes #13068
* Closes #13069
* Closes #13070
* Closes #13071

### CODE REVIEW

File: warehouse/macaroons/caveats/_core.py [hunk 1]
```diff
@@ -111,7 +111,13 @@ def lookup(self, /, tag: int) -> type[Caveat] | None:
...
 
 
-def as_caveat(*, tag: int) -> Callable[[type[T]], type[T]]:
+# TODO: The return signature detected is `"Union[Type[Dataclass], DataclassProxy]"`,
+# but the expectation is `Type[Dataclass]`.
+# See https://github.com/pydantic/pydantic/issues/4498 but not exactly the same.
+# This might not be corrected in pydantic until 2.0.
+# Original signature with type hints:
+# def as_caveat(*, tag: int) -> Callable[[type[T]], type[T]]:
+def as_caveat(*, tag: int):
```
Comment (miketheman) [inline warehouse/macaroons/caveats/_core.py:120]: @dstufft Maybe you can help here? I got lost down a rabbit hole before giving up and writing this comment.

File: tests/unit/rate_limiting/test_core.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def test_resets_in_expired(self, metrics):
             ]
         )
 
-        limiter._window = pretend.stub(get_window_stats=lambda l, *a: next(stats))
+        limiter._window = pretend.stub(get_window_stats=lambda L, *a: next(stats))
```
Comment (di) [inline tests/unit/rate_limiting/test_core.py:122]: @miketheman What bump introduced this? I'm curious why this is happening only on a few lines when we probably have lots of other instances of this.

Comment (ewdurbin) [reply tests/unit/rate_limiting/test_core.py:122]: flake8 went from 4.0.0 to 6.0.0, guessing its that.

Comment (miketheman) [reply tests/unit/rate_limiting/test_core.py:122]: Yep - links to issue in [`3c78893` (#13075)](https://github.com/pypi/warehouse/pull/13075/commits/3c7889309c189555dc221e54b7a2d7c1f8b90712)

Comment (ewdurbin) [reply tests/unit/rate_limiting/test_core.py:122]: https://github.com/PyCQA/pycodestyle/pull/1106/files

Comment (ewdurbin) [reply tests/unit/rate_limiting/test_core.py:122]: "ambiguous identifier" rule now applies to lambdas with the pycodestyle bump.

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**